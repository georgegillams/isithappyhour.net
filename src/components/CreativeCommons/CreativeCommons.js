import React from 'react';

import Paragraph from '@george-gillams/components/paragraph';
import TextLink from 'components/common/TextLink';
import { Background, Inner, Wrapper } from './creative-commons.syles';

const CreativeCommons = props => {
  const { ...rest } = props;

  return (
    <Wrapper {...rest}>
      <Background />
      <Inner anchor={false} name="Copyright">
        <Paragraph>
          Most of my photos are licensed under{' '}
          <TextLink hrefExternal href="https://creativecommons.org/licenses/by-sa/3.0/">
            Creative Commons BY-SA 3.0
          </TextLink>
          .<br />
          If you are unsure about your right to use them please <TextLink href="/contact">contact me</TextLink>.
        </Paragraph>
      </Inner>
    </Wrapper>
  );
};

export default CreativeCommons;
