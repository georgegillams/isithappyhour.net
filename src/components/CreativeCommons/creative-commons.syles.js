import styled from 'styled-components';
import { spacingBase, spacingLg } from '@george-gillams/components/constants/layout';
import {
  alternatingBackgroundColor1,
  alternatingBackgroundColor1DarkMode,
} from '@george-gillams/components/constants/colors';
import Subsection from '@george-gillams/components/subsection';

export const Wrapper = styled.div`
  position: relative;
  display: flex;
  width: 100%;
  margin-top: ${spacingLg};
  margin-left: 0;
  padding-top: ${spacingBase};
  padding-bottom: ${spacingLg};
  justify-content: center;
`;

export const Background = styled.div`
  position: absolute;
  top: 0;
  left: calc(50% - 50vw);
  z-index: -1;
  width: 100vw;
  height: 100%;
  background-color: ${alternatingBackgroundColor1};

  @media (prefers-color-scheme: dark) {
    background-color: ${alternatingBackgroundColor1DarkMode};
  }
`;

export const Inner = styled(Subsection)`
  width: 100%;
  padding-right: 1.875rem;
  padding-left: 1.865rem;
  text-align: center;

  @include bpk-breakpoint-tablet {
    padding-right: 0.9375rem;
    padding-left: 0.9375rem;
  }
`;
