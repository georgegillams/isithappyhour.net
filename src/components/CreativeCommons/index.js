import CreativeCommons from './CreativeCommons';

export default CreativeCommons;
export { CreativeCommons };
