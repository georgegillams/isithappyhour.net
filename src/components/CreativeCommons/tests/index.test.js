import React from 'react';

import { render } from '@testing-library/react';

import CreativeCommons from '..';

describe('<CreativeCommons />', () => {
  it('Should render correctly', () => {
    const { container } = render(<CreativeCommons />);

    expect(container).toMatchSnapshot();
  });
  it('Should render correctly with custom className', () => {
    const { container } = render(<CreativeCommons className="test-custom-className" />);

    expect(container).toMatchSnapshot();
  });
});
