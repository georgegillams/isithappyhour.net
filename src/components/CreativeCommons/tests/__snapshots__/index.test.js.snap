// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CreativeCommons /> Should render correctly 1`] = `
<div>
  <div
    class="creative-commonssyles__Wrapper-sc-qff226-0 gOCDiS"
  >
    <div
      class="creative-commonssyles__Background-sc-qff226-1 ddrRBh"
    />
    <div
      class="sc-kDDrLX iNAFzC creative-commonssyles__Inner-sc-qff226-2 gkGslB"
    >
      <h1
        class="sc-hKMtZM juqKJr sc-evZas goeUxJ"
        id="copyright"
      >
        Copyright
      </h1>
      <span
        class="sc-bczRLJ bhVmoO"
      >
        Most of my photos are licensed under
         
        <a
          aria-label="Creative Commons BY-SA 3.0 (opens in a new tab)"
          class="sc-ftvSup cMmkUr"
          href="https://creativecommons.org/licenses/by-sa/3.0/"
          rel="noopener noreferrer"
          target="_blank"
        >
          Creative Commons BY-SA 3.0
          <div
            class="sc-papXJ ecwnCf"
          >
            <svg
              class="sc-iBkjds eZahvc sc-jqUVSM fjeMmK"
              height="18"
              viewBox="0 0 24 24"
              width="18"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M12.5 11.3c.4.4 1 .4 1.4 0L19 6.2V9c0 .6.4 1 1 1s1-.4 1-1V4c0-.6-.4-1-1-1h-5c-.6 0-1 .4-1 1s.4 1 1 1h2.4l-4.9 4.9c-.4.4-.4 1 0 1.4zM19 19c0 1.1-.9 2-2 2H5c-1.1 0-2-.9-2-2V7c0-1.1.9-2 2-2h7v2H5v12h12v-7h2"
              />
            </svg>
          </div>
        </a>
        .
        <br />
        If you are unsure about your right to use them please 
        <a
          class="sc-ftvSup cMmkUr"
          href="/contact"
        >
          contact me
        </a>
        .
      </span>
    </div>
  </div>
</div>
`;

exports[`<CreativeCommons /> Should render correctly with custom className 1`] = `
<div>
  <div
    class="creative-commonssyles__Wrapper-sc-qff226-0 gOCDiS test-custom-className"
  >
    <div
      class="creative-commonssyles__Background-sc-qff226-1 ddrRBh"
    />
    <div
      class="sc-kDDrLX iNAFzC creative-commonssyles__Inner-sc-qff226-2 gkGslB"
    >
      <h1
        class="sc-hKMtZM juqKJr sc-evZas goeUxJ"
        id="copyright"
      >
        Copyright
      </h1>
      <span
        class="sc-bczRLJ bhVmoO"
      >
        Most of my photos are licensed under
         
        <a
          aria-label="Creative Commons BY-SA 3.0 (opens in a new tab)"
          class="sc-ftvSup cMmkUr"
          href="https://creativecommons.org/licenses/by-sa/3.0/"
          rel="noopener noreferrer"
          target="_blank"
        >
          Creative Commons BY-SA 3.0
          <div
            class="sc-papXJ ecwnCf"
          >
            <svg
              class="sc-iBkjds eZahvc sc-jqUVSM fjeMmK"
              height="18"
              viewBox="0 0 24 24"
              width="18"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M12.5 11.3c.4.4 1 .4 1.4 0L19 6.2V9c0 .6.4 1 1 1s1-.4 1-1V4c0-.6-.4-1-1-1h-5c-.6 0-1 .4-1 1s.4 1 1 1h2.4l-4.9 4.9c-.4.4-.4 1 0 1.4zM19 19c0 1.1-.9 2-2 2H5c-1.1 0-2-.9-2-2V7c0-1.1.9-2 2-2h7v2H5v12h12v-7h2"
              />
            </svg>
          </div>
        </a>
        .
        <br />
        If you are unsure about your right to use them please 
        <a
          class="sc-ftvSup cMmkUr"
          href="/contact"
        >
          contact me
        </a>
        .
      </span>
    </div>
  </div>
</div>
`;
